hw/ip/spi_device/doc/_index.md

```diff
@@ -309,6 +361,101 @@ All reads and writes to/from the SRAM for RXF and TXF activity are managed by
 direct reads and writes through the TLUL bus interface, managed by the
 auto-generated register file control logic.
 
+## SPI Flash and Passthrough Modes
+
+### Command Information List
+
+The SW may configure the map from the received opcode to the command process module by programming *cmd_info* list.
+Current SPI_DEVICE provides 24 command information entries.
+Each entry represents a command.
+Details of the fields are explained in the {{<regref "CMD_INFO_0">}}
+
+First 11 commands are assigned to specific submodules.
+
+Index  | Assigned Submodule
+-------|--------------------
+[2:0]  | Read Status
+[3]    | Read JEDEC ID
+[4]    | Read SFDP
+[10:5] | Read commands
+
+When Read Status, Read JEDEC ID commands are processed in Flash mode, the other fields in the command information entry are not used.
+The submodules directly return data on the MISO line (SD[1]).
+In Passthrough mode, if Read Status and Read JEDEC ID commands are intercepted by the internal HW,
+the other fields in the command information entries are ignored also.
+
+The main use of the fields other than *opcode* and *valid* is to control the output enable in the passthrough logic.
+See [Output Enable Control](#output-enable-control) section for more.
+
+*upload* and *busy* fields are used in the SPI Flash/ Passthrough modes.
+See [Command Upload](#command-upload) section for details.
+
+### Command Parser
+
+![Command Parser block](cmdparse.svg)
+
+Command parser (*cmdparse*) processes the first byte of the SPI and activates the processing submodules depending on the received opcode and the *cmd_info* list described in the previous section.
+
+The cmdparse compares the recevied opcode with the *cmd_info.opcode* data structure.
+If any entry matches to the received opcode, the cmdparse hands over the matched command information entry with the index to the corresponding submodule.
+As explained in the [previous section](#command-information-list), the command parser checks the index to activate Read Status / Read JEDEC ID/ Read Command / Address 4B modules.
+Other than the first 11 slots and last two slots (the last two slots are not visible to SW), the cmdparse checks the *upload* field and activates the upload module if the field is set.
+
+SW may configure the submodules if the modules process the command in the passthrough mode or not by changing {{<regref "INTERCEPT_EN">}} CSR.
```

Comment:
nit: should we slightly rephrase to something like

`SW can configure whether a submodule should process the command while in passthrough mode by setting the {{<regref "INTERCEPT_EN">}} CSR accordingly.`?